LEGO1/tgl/d3drm/mesh.cpp

```diff
@@ -0,0 +1,166 @@
+#include "impl.h"
+
+using namespace TglImpl;
+
+DECOMP_SIZE_ASSERT(D3DRMVERTEX, 0x24);
+
+DECOMP_SIZE_ASSERT(Mesh, 0x4);
+DECOMP_SIZE_ASSERT(MeshImpl, 0x8);
+
+// Inlined only
+MeshImpl::~MeshImpl()
+{
+	if (m_data) {
+		free(m_data);
```

Comment:
This should be `delete`

Reply:
I don't think that's the case.

A delete call includes a null check, but not a set to null. If this were actually a delete call then the code would either end up with two nested if statements rather than a single one, or the code setting to null outside of the if statement.

MeshImpl is the odd one out, not following the pattern the others have of directly holding a Direct3DRM handle, so it's possible the people who came in later and made that change did use malloc / free for the holder struct instead of new / delete like the initial Tgl implementers did in the leak.